packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -245,7 +245,7 @@ export class Rule extends Resource implements IRule {
           });
           new CfnEventBusPolicy(eventBusPolicyStack, 'GivePermToOtherAccount', {
             action: 'events:PutEvents',
-            statementId: `Allow-account-${sourceAccount}`,
+            statementId: `Allow-account-${sourceAccount}-${Names.uniqueId(this)}`,
```

Comment:
What is the maximum length of a Sid? Can you ensure this length never grows too large?

Reply:
`this.node.addr` may be better.